diff --git a/src/common/mythread.h b/src/common/mythread.h
index 37b5216..0819aae 100644
--- a/src/common/mythread.h
+++ b/src/common/mythread.h
@@ -337,8 +337,10 @@ mythread_condtime_set(mythread_condtime *condtime, const mythread_cond *cond,
 
 #define WIN32_LEAN_AND_MEAN
 #ifdef MYTHREAD_VISTA
-#	undef _WIN32_WINNT
-#	define _WIN32_WINNT 0x0600
+#   if !defined(WINAPI_FAMILY) || !(WINAPI_FAMILY==WINAPI_FAMILY_PC_APP || WINAPI_FAMILY==WINAPI_FAMILY_PHONE_APP)
+#	    undef _WIN32_WINNT
+#	    define _WIN32_WINNT 0x0600
+#   endif
 #endif
 #include <windows.h>
 #include <process.h>
diff --git a/src/common/tuklib_cpucores.c b/src/common/tuklib_cpucores.c
index bb3f2f7..76f30a8 100644
--- a/src/common/tuklib_cpucores.c
+++ b/src/common/tuklib_cpucores.c
@@ -14,7 +14,9 @@
 
 #if defined(_WIN32) || defined(__CYGWIN__)
 #	ifndef _WIN32_WINNT
-#		define _WIN32_WINNT 0x0500
+#       if !defined(WINAPI_FAMILY) || !(WINAPI_FAMILY==WINAPI_FAMILY_PC_APP || WINAPI_FAMILY==WINAPI_FAMILY_PHONE_APP)
+#		    define _WIN32_WINNT 0x0500
+#       endif
 #	endif
 #	include <windows.h>
 
diff --git a/src/common/tuklib_physmem.c b/src/common/tuklib_physmem.c
index a1bccb2..bd99bbe 100644
--- a/src/common/tuklib_physmem.c
+++ b/src/common/tuklib_physmem.c
@@ -17,7 +17,9 @@
 // gives wrong results (from our point of view).
 #if defined(_WIN32) || defined(__CYGWIN__)
 #	ifndef _WIN32_WINNT
-#		define _WIN32_WINNT 0x0500
+#       if !defined(WINAPI_FAMILY) || !(WINAPI_FAMILY==WINAPI_FAMILY_PC_APP || WINAPI_FAMILY==WINAPI_FAMILY_PHONE_APP)
+#		    define _WIN32_WINNT 0x0500
+#       endif
 #	endif
 #	include <windows.h>
 
@@ -79,6 +81,13 @@ tuklib_physmem(void)
 	uint64_t ret = 0;
 
 #if defined(_WIN32) || defined(__CYGWIN__)
+#include <winapifamily.h>
+#if (_WIN32_WINNT >= 0x501) || (defined(WINAPI_FAMILY_PARTITION) && WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP) && !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP))
+	MEMORYSTATUSEX meminfo;
+	meminfo.dwLength = sizeof(meminfo);
+	if (GlobalMemoryStatusEx(&meminfo))
+		ret = meminfo.ullTotalPhys;
+#else
 	if ((GetVersion() & 0xFF) >= 5) {
 		// Windows 2000 and later have GlobalMemoryStatusEx() which
 		// supports reporting values greater than 4 GiB. To keep the
@@ -107,6 +116,7 @@ tuklib_physmem(void)
 		GlobalMemoryStatus(&meminfo);
 		ret = meminfo.dwTotalPhys;
 	}
+#endif
 
 #elif defined(__OS2__)
 	unsigned long mem;
